### PR TITLE
NFC-386 Add new error handling

### PR DESCRIPTION
**Pull request by laurageiger-mohemian**
## JIRA
- https://airside.atlassian.net/browse/NFC-386

## Description 
A TimeOutError was added to the NFCPassportReaderError. Moreover, the thrown errors are now handled in the right way for a certain behavior. 

NFC SDK PR: https://github.com/airsidemobile/nfc-sdk-ios/pull/100#issue-2271405965
Habicht PR: https://github.com/airsidemobile/habicht-ios/pull/1931#issue-2271386864
RFC: https://airside.atlassian.net/l/cp/h9EHvXd5

### CODE REVIEW

**Review by daniel-moh — COMMENTED** [review]
@laurageiger-mohemian could you remove the formatting changes please, it's a bit hard to spot the actual changes otherwise. Let's stick with the formatting of the upstream repo. Will make it easier to stay in sync with upstream going forward.

File: Sources/NFCPassportReader/PassportReader.swift
Comment (daniel-moh) [inline Sources/NFCPassportReader/PassportReader.swift:137]:
```suggestion
                    Log.error("     - Session timeout")
```

File: Sources/NFCPassportReader/PassportReader.swift
Comment (daniel-moh) [inline Sources/NFCPassportReader/PassportReader.swift:209]: What error case is this exactly? 🤔

**Review by daniel-moh — COMMENTED** [review]
Nice! One question (see inline).

File: Sources/NFCPassportReader/PassportReader.swift
Comment (laurageiger-mohemian) [reply Sources/NFCPassportReader/PassportReader.swift:209]: Good question. The explanation of this error is:

```
// The tag has responded with an error.
static var readerTransceiveErrorTagResponseError: NFCReaderError.Code 
```

However, the following error would **suit better** according to its description:

```
// The reader lost the connection to the tag.
static var readerTransceiveErrorTagConnectionLost: NFCReaderError.Code

```

The NFCReaderError from the first code block was thrown by the CoreNFC lib when the connection was lost to the chip. This is why I chose this error.

Comment (daniel-moh) [reply Sources/NFCPassportReader/PassportReader.swift:209]: _We discussed offline:_

- `readerTransceiveErrorTagResponseError` is thrown when we force a "connection lost" scenario: moving the phone away from the passport. Let's document this inline though since it is not clear just from reading the code.
- Also include `NFCReaderTransceiveErrorTagConnectionLost` in the mapping (even though it might not be reproducible for now but from the naming it is quite clear that it also makes sense to map to a `NFCPassportReaderError.ConnectionError`).

**Review by daniel-moh — APPROVED** [review]
LGTM. ✅ 

Let's not merge this PR here but submit it to the [upstream repo](https://github.com/AndyQ/NFCPassportReader). Once they merge it we can then sync our fork with theirs and integrate from our fork's master branch.

These are independent tasks thought we don't need to wait for them to be done. In the meantime we can use the `NFC-386-fix-thrown-errors-on-nfc-scan` branch of this repo int the NFC SDK, as discussed.

- Story to submit the change to the upstream repo: https://airside.atlassian.net/browse/NFC-389
- Story to sync our fork and integrate the latest changes to the NFC SDK: https://airside.atlassian.net/browse/NFC-390

**Pull request closed without merging**
